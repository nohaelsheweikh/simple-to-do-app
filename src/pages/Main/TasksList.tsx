import React ,{useEffect,useState}from "react";
import * as styles from "../../styles/card"
import { 
    Container,
    CardsListWrapper,
    HiddenButton,
    StyledInput,
    SearchWrapper
  } from "../../styles/tasks"
import history from "../../utils/history"
import { IoMdTrash,IoMdAdd} from 'react-icons/io';

  

const TasksList = (props:any) =>{
  const [tasks,setTasks] = useState([props.tasks]);
  useEffect(() => setTasks(props.tasks), [props.tasks]);


  if(!props.tasksList){
    return <h4>is loading</h4>
  }
  return (
    <Container>
       <styles.CreateButtonWrapper>
         <styles.CreateButton onClick={() => history.push('./create')}>Create Category</styles.CreateButton>
       </styles.CreateButtonWrapper> 

      
          <form onSubmit={props.fetchTasksByDate}>
          <HiddenButton>Search</HiddenButton> 
            <SearchWrapper>
              <StyledInput type="date" onChange={props.onChange} name="date"/>
            </SearchWrapper>
          </form>

      <CardsListWrapper>
        <styles.CardContainer>  
        {
          tasks||props.tasksList.map((category:any, index:number) => {
            
              return  (
              <styles.Card key={index} >
                <styles.ADDTaskButton>
                  <IoMdAdd 
                   color={'green'} 
                   onClick={() =>history.push(`/task/${category.id}`)}
                   />
                </styles.ADDTaskButton>
                  <styles.Title>{category.category}</styles.Title>
                      {category.tasks.map((task:any, index:number) => {
                    return(
                        <div key={index}>
                         
                          <styles.Date>
                          <IoMdTrash 
                          color={"red"}
                          onClick={() => props.deleteTask(category.id,task.id)}      
                            />{task.date}</styles.Date>
                          <styles.Description>{task.name}</styles.Description>
                      </div>

                      )}
                    )}
                
              </styles.Card>
          )})}
        </styles.CardContainer>
      </CardsListWrapper>
  </Container>
  )
};

export default TasksList;
