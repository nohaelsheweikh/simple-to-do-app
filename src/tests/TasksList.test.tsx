import React from 'react';
import { configure,mount } from 'enzyme';
import TasksList from '../pages/Main/TasksList';
import Adapter from 'enzyme-adapter-react-16'
import configureMockStore from "redux-mock-store";
import thunk from 'redux-thunk';
import toJson from "enzyme-to-json";
import {Provider} from 'react-redux'

configure({ adapter: new Adapter() })
const mockStore = configureMockStore([thunk]);
const store = mockStore({})
    
    const tasks =  [
        {
          "id": 1,
          "category": {
            "name": "sport"
          },
          "task_name": "tennis"
        },
        {
          "id": 2,
          "category": {
            "name": "work"
          },
          "task_name": "dev"
        },
        {
          "id": 1576342806617.201,
          "category": {
            "name": "Shopping"
          },
          "task_name": "milk,eggs"
        }
      ]
    
    describe('TaskCards component renders a list of Cards', () => {
      it('renders correctly', () => {
        const wrapper  = mount(
                <Provider store={store}>
                    <TasksList
                        tasks={tasks}
                    />
                </Provider>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
      });  
 })
 