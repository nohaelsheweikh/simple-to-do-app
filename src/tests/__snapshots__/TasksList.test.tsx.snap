// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCards component renders a list of Cards renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <TasksList
    tasks={
      Array [
        Object {
          "category": Object {
            "name": "sport",
          },
          "id": 1,
          "task_name": "tennis",
        },
        Object {
          "category": Object {
            "name": "work",
          },
          "id": 2,
          "task_name": "dev",
        },
        Object {
          "category": Object {
            "name": "Shopping",
          },
          "id": 1576342806617.201,
          "task_name": "milk,eggs",
        },
      ]
    }
  >
    <h4>
      is loading
    </h4>
  </TasksList>
</Provider>
`;

exports[`TasksList component renders a list of Cards renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Card
    tasks={
      Array [
        Object {
          "category": Object {
            "name": "sport",
          },
          "id": 1,
          "task_name": "tennis",
        },
        Object {
          "category": Object {
            "name": "work",
          },
          "id": 2,
          "task_name": "dev",
        },
        Object {
          "category": Object {
            "name": "Shopping",
          },
          "id": 1576342806617.201,
          "task_name": "milk,eggs",
        },
      ]
    }
  >
    <h4>
      is loading
    </h4>
  </Card>
</Provider>
`;
